src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2614,8 +2702,9 @@ private void AddDeclaredNontypeMembers(MembersAndInitializersBuilder builder, Di
                 if (builder.RecordDeclarationWithParameters is null)
                 {
                     builder.RecordDeclarationWithParameters = syntax;
-                    builder.InstanceInitializersForRecordDeclarationWithParameters = new ArrayBuilder<FieldOrPropertyInitializer.Builder>();
-                    return builder.InstanceInitializersForRecordDeclarationWithParameters;
+                    // Keep track of which instance initializers we'll want to update when we synthesize record properties
+                    builder.InstanceInitializersIndexForRecordDeclarationWithParameters = builder.InstanceInitializers.Count;
```

Comment:
>builder.InstanceInitializersIndexForRecordDeclarationWithParameters = builder.InstanceInitializers.Count; [](start = 20, length = 105)

This feels more fragile now. If we merge this PR, let's open a follow-up issue to avoid pre-allocating a builder that might not be used after all.

Reply:
@AlekseyTs Given that we're using pooled instances now. Do you still want a follow-up issue opened?

---
In reply to: [562274378](https://github.com/dotnet/roslyn/pull/50650#discussion_r562274378) [](ancestors = 562274378)

Reply:
> Given that we're using pooled instances now. Do you still want a follow-up issue opened?

Yes

---
In reply to: [564152951](https://github.com/dotnet/roslyn/pull/50650#discussion_r564152951) [](ancestors = 564152951,562274378)

Reply:
Filed https://github.com/dotnet/roslyn/issues/50775